src/Generator/Passes/GenerateSymbolsPass.cs

```diff
@@ -53,7 +53,38 @@ private void GenerateSymbols()
                 var e = new SymbolsCodeEventArgs(module);
                 SymbolsCodeGenerated?.Invoke(this, e);
                 if (string.IsNullOrEmpty(e.CustomCompiler))
+                {
+                    if (module == Options.SystemModule ||
```

Comment:
Please factor this code better, right now it introduces a lot of nesting.

Reply:
I'll see what I can do.

Reply:
Done.